apps/omg/lib/omg/ethereum_event_listener.ex

```diff
@@ -84,42 +109,56 @@ defmodule OMG.EthereumEventListener do
 
     _ = Logger.info("Started #{inspect(__MODULE__)} for #{service_name}, synced_height: #{inspect(height_to_check_in)}")
 
-    {:noreply, {initial_state, callbacks_map}}
+    {:noreply, {initial_state, callbacks}}
   end
 
-  def handle_info(:send_metrics, {core, _callbacks} = state) do
-    :ok = :telemetry.execute([:process, __MODULE__], %{}, core)
-    {:noreply, state}
+  def handle_info(:send_metrics, {%Core{} = state, callbacks}) do
+    :ok = :telemetry.execute([:process, __MODULE__], %{}, state)
+    {:noreply, {state, callbacks}}
```

Comment:
Actually, the reason why I pointed this out is because I was worried about performance. In the back of my head I remembered a PR from Jose that optimised some erlang code by "not recreating" the tuple. Now I've read https://github.com/erlang/otp/pull/1080 and http://blog.erlang.org/core-erlang-wrapup/ which seems to point out that this optimisation is now recognized by the compiler! Horaaay!